content/posts/unpublished/migration-from-karma-jasmine-to-jest/post.md

```diff
@@ -0,0 +1,139 @@
+# Migration from Karma/Jasmine to Jest
+
+Given the faster execution, clearer stack trace of Jest, Karma is being deprecated; decided to migrate to Jest.
+
+![](assets/migration_karma_jest.png)
+
+Authors: Utku, Omar Rojas
+Date: unpublished
+Category: frontend
+
+tags: web,testing,test,unit,frontend,karma,jest,jasmine
+
+---
+
+## Introduction
+
+In Angular 16, Karma was deprecated while Jest was introduced in experimental mode. The Angular team moved towards Jest as the recommended testing framework for Angular applications. While Karma was the default testing framework for Angular in the past, Jest is being explored as a potential replacement due to its simplicity, speed, and built-in features like snapshot testing and mocking. However, it’s important to note that Jest is still in the experimental stage in Angular 16 and Angular 17, and using it in production may have limitations or unsupported features.
+
+Given the faster execution, clearer stack trace of Jest, Karma is being deprecated; decided to migrate to Jest.
+
+## Migration
+
+Below steps have been followed for complete migration in retail(universal) web app:
+
+1. Delete the tests properties inside targets object completely in **project.json** files.
+
+2. Delete all the references to **karma.conf.js** files by globally searching in your project.
+
+3. Delete all **karma.conf.js** files.
+
+4. Delete all **test.ts** files.
+
+5. Delete all Karma and Jasmine related dependencies from **package.json**.
+
+6. Run `npm i` to update **package-lock.json** and **node_modules**.
+
+7. Go to [@nx/jest | Nx](https://nx.dev/nx-api/jest) to follow the instructions below:
+
+    1. Do not install the **nx/jest** unless you don’t already have it in **package.json**.
+
+    2. To add Jest to projects, run `nx g @nx/jest:configuration --project=<project-name>` for all projects in the repository one by one except e2e and assets projects. This command will add required config to **project.json** files as well as making changes in **tsconfig.spec.json** and **jest.config.ts** files per project. This will allow Nx to run tests properly. Content of **jest.config.ts**:
+
+    ```typescript
+    export default {
+      displayName: 'retail-universal',
```

Comment:
Perhaps give an example name such as "demo-angular" and let the reader know that this is the name of the angular application

Reply:
done!